plugins/logs/plugin.go

```diff
@@ -894,6 +927,62 @@ func (p *Plugin) maskEvent(ctx context.Context, txn storage.Transaction, event *
 	return nil
 }
 
+func (p *Plugin) dropEvent(ctx context.Context, txn storage.Transaction, event *EventV1) (bool, error) {
+
+	drop, err := func() (rego.PreparedEvalQuery, error) {
+
+		p.dropMutex.Lock()
+		defer p.dropMutex.Unlock()
+
+		if p.drop == nil {
+
+			query := ast.NewBody(ast.NewExpr(ast.NewTerm(p.config.dropDecisionRef)))
+
+			r := rego.New(
+				rego.ParsedQuery(query),
+				rego.Compiler(p.manager.GetCompiler()),
+				rego.Store(p.manager.Store),
+				rego.Transaction(txn),
+				rego.Runtime(p.manager.Info),
+				rego.EnablePrintStatements(p.manager.EnablePrintStatements()),
+				rego.PrintHook(p.manager.PrintHook()),
+			)
+
+			pq, err := r.PrepareForEval(context.Background())
+			if err != nil {
+				return rego.PreparedEvalQuery{}, err
+			}
+
+			p.drop = &pq
+		}
+
+		return *p.drop, nil
+	}()
+
+	if err != nil {
+		return false, err
+	}
+
+	input, err := event.AST()
+	if err != nil {
+		return false, err
+	}
+
+	rs, err := drop.Eval(
+		ctx,
+		rego.EvalParsedInput(input),
+		rego.EvalTransaction(txn),
+	)
+
+	if err != nil {
+		return false, err
+	} else if len(rs) == 0 {
+		return false, nil
+	}
+
+	return true, nil
```

Comment:
I'd guess that the `true` should be more elaborate as this would be true for all results that are defined also related to the other comment on the docs. Either use `rs.Allowed()` https://github.com/open-policy-agent/opa/blob/d75bbdd0f9d5600f3c3d87b6f80d525a62daebd1/rego/resultset.go#L81 or explicitly check that the conditions (just one result, it must be boolean) are fulfilled and err otherwise. A rule like `drop = "bogus" { true }` would otherwise be silently ignored...

Reply:
Heh, I had thought about suggesting `Allowed()`, too. It's useful, I guess, but a little bit weird since it's coming with so many connotations...